### DISCUSSION TITLE
Different result with different images in "compose" function

### DISCUSSION THREAD

**Discussion by Archive-d** [ImageMagick/ImageMagick#5288] [Help] [answered]
I am trying to compose the following images

![Acorn](https://user-images.githubusercontent.com/76863644/176997920-2ab9960c-839a-40e0-a52b-531535e76ef9.png)
![Beach](https://user-images.githubusercontent.com/76863644/176997934-5544075c-903e-4eda-bfb2-b95473ebea85.png)
![Purple](https://user-images.githubusercontent.com/76863644/176997953-6b421654-a705-4d58-beb9-12e4af4eaf09.png)

using
`magick Acorn.png Purple.png -compose lighten -composite Beach.png -compose darken -composite out.png`

To Achieve

![New Project](https://user-images.githubusercontent.com/76863644/176998213-ccde9b6c-8f5f-4c0f-96bc-672b0c9fe432.png)

Not working.

However when using this image it works
![76](https://user-images.githubusercontent.com/76863644/176998293-4850883e-71f2-4a08-841c-67b721a1083f.png)

P.s, I am working with multiple images so the code is,
`for %%A in ("D:\Icons\MyIcons\*") do (
  for %%B in ("D:\Icons\Pack\Dual_Color\Beach_Purple\Purple.png") do (
        magick "%%A" "%%B" -compose screen -composite "D:\Icons\Pack\Dual_Color\Beach_Purple\Beach.png" -compose darken -composite "D:\Icons\Pack\Dual_Color\Beach_Purple\Icons\%%~nA.png"
  )
)`

OS. Windows 10
Imagemagick 7

Also, Am any spaces in folder names result in an error even after using """"

Thank you

**Comment by snibgo** [comment]
> Not working.

Diagnosing the problem from that brief explanation is difficult.

If the output is monochrome, the problem is that your first input is monochrome. Use "-set colorspace sRGB" like this:
```
magick Acorn.png -set colorspace sRGB Purple.png -compose lighten -composite Beach.png -compose darken -composite out.png
```

> Also, Am any spaces in folder names result in an error even after using """"

What is the error? Please paste the command, with the full error text.

**Comment by Archive-d** [comment, marked as answer]
Hey Snibgo,

doing `-set colorspace sRGB` worked 😄 
What did it do exactly?

For Folder names with spaces the error I get is 
`magick: unable to open image 'D:\Icons\Pack\Dual_Color\Beach': No such file or directory @ error/blob.c/OpenBlob/3533.`

code used
```
for %%A in ("D:\Icons\MyIcons\*") do (
  for %%B in ("D:\Icons\Pack\Dual_Color\Beach Purple\Purple.png") do ( magick
``` operation...

**Comment by snibgo** [comment]
"-composite" takes 2 or 3 input images and makes one output. The metadata of the output comes from the metadata of the first input.

What is the metadata of acorn.png and 76.png?
```
f:\web\im>%IMG7%magick acorn.png info:
acorn.png PNG 1024x1024 1024x1024+0+0 8-bit Gray 256c 19161B 0.000u 0:00.002

f:\web\im>%IMG7%magick 76.png info:
76.png PNG 1024x1024 1024x1024+0+0 8-bit sRGB 17c 6883B 0.000u 0:00.002
```
acorn.png is grayscale. When you use that as the first input, the output is also grayscale.

"-set colorspace sRGB" changes the colorspace to be colour, so the first input to "-composite" is colour, so the output is also colour.

The OP has:
```
"D:\Icons\Pack\Dual_Color\Beach_Purple\Beach.png"
```
... but your last post has:

```
"D:\Icons\Pack\Dual_Color\Beach Purple\Purple.png"
```
So there is a space instead of an underscore. You need to use the correct name, whichever it is.

Personally, I don't like spaces in directory or file names. They cause more problems than they solve. But IM and other programs work correctly, provided the names are quoted.

**Reply by Archive-d** [reply]
Thank you for the explanation.

I changed the folder name and then directory to 
`"D:\Icons\Pack\Dual_Color\Beach Purple\Purple.png"`

to get the error I was explaining. I changed the code accordingly too.

I will follow your advice on not using Spaces for better results and less problem.

Thank you.
